python/lsst/atmospec/formatters.py

```diff
@@ -31,10 +33,27 @@ class SpectractorSpectrumFormatter(FileFormatter):
     unsupportedParameters = None
 
     def _readFile(self, path, pytype=None):
-        return Spectrum(path)
+        spectrogramPath = path + 'spectrogram.fits'
+        psfPath = path + 'table.csv'
+        # TODO: Add round-tripping of detected lines with lines.csv once
+        # Spectractor supports that
+        if os.path.isfile(spectrogramPath) and os.path.isfile(psfPath):
+            return Spectrum(path, fast_load=False,
+                            spectrogram_file_name_override=spectrogramPath,
+                            psf_file_name_override=psfPath)
+        else:
+            return Spectrum(path, fast_load=True)
 
     def _writeFile(self, inMemoryDataset):
+        spectrogramFilename = self.fileDescriptor.location.path + 'spectrogram.fits'
+        linesFilename = self.fileDescriptor.location.path + 'lines.csv'
+        psfFilename = self.fileDescriptor.location.path + 'table.csv'
```

Comment:
Um. Hang on. This code fundamentally breaks the way that butler formatters work. It's never going to work with an object store. A formatter can only write one file or else it is a disassembled composite storage class where butler knows that multiple files are going to be written out and where they are. @mfisherlevine we are going to have to reorganize all of this and think deeply.